Wikipedia/Code/SearchLanguagesBarViewController.swift

```diff
@@ -0,0 +1,145 @@
+import Foundation
+
+@objc protocol SearchLanguagesBarViewControllerDelegate: class {
+    func searchLanguagesBarViewController(controller: SearchLanguagesBarViewController, didChangeCurrentlySelectedSearchLanguage language: MWKLanguageLink)
+}
+
+class SearchLanguagesBarViewController: UIViewController, WMFPreferredLanguagesViewControllerDelegate {
+    weak var delegate: SearchLanguagesBarViewControllerDelegate?
+    
+    @IBOutlet private var languageButtons: [UIButton] = []
+    @IBOutlet private var otherLanguagesButton: UIButton?
+    @IBOutlet private var heightConstraint: NSLayoutConstraint?
+    
+    private var previousFirstLanguage: MWKLanguageLink?
+    private var hidden: Bool = false {
+        didSet {
+            if(hidden){
+                heightConstraint!.constant = 0
+                view.hidden = true
+            }else{
+                heightConstraint!.constant = 44
+                view.hidden = false
+            }
+        }
+    }
+
+    private var _currentlySelectedSearchLanguage: MWKLanguageLink?
+    private(set) var currentlySelectedSearchLanguage: MWKLanguageLink? {
+        get {
+            return _currentlySelectedSearchLanguage
+        }
+        set {
+            _currentlySelectedSearchLanguage = newValue
+            NSUserDefaults.wmf_userDefaults().wmf_setCurrentSearchLanguageDomain(newValue?.siteURL())
+            delegate?.searchLanguagesBarViewController(self, didChangeCurrentlySelectedSearchLanguage: newValue!)
+            updateLanguageBarLanguageButtons()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        _ = languageButtons.map{$0.tintColor = UIColor.wmf_blueTintColor()}
```

Comment:
In this case, yes… one of the things about swift is that it is functional… when you want to transform an object not just iterate, it provides you a map function to make your intentions clear.

This case is just the opposite, you really just want to enumerate, and it would be best to use the method that clearly identifies that intent of just iterating.